dev/public/static/js/elements-identification.js

```diff
@@ -98,60 +108,95 @@ function updateStyleByAPI(bpmnIds, bpmnKind) {
         break;
     }
   }
-
-  updateStyle(bpmnIds, style);
+  return style;
 }
 
-function resetStyleByAPI() {
-  resetStyle(lastIdentifiedBpmnIds);
-}
+function updateStyleByAPI(bpmnIds, bpmnKind) {
+  const subProcessChildrenIds = bpmnIds.filter(isChildOfSubProcess);
+  const otherIds = bpmnIds.filter(bpmnId => !subProcessChildrenIds.includes(bpmnId));
 
-function updateSelectedBPMNElements(textArea, bpmnKind) {
-  log(`Searching for Bpmn elements of '${bpmnKind}' kind`);
-  const elementsByKinds = getElementsByKinds(bpmnKind);
+  if (subProcessChildrenIds.length > 0) {
+    styledPoolAndLaneIds = getElementsByKinds([ShapeBpmnElementKind.POOL, ShapeBpmnElementKind.LANE]).map(element => element.bpmnSemantic.id);
+    updateStyle(styledPoolAndLaneIds, { opacity: 5, font: { color: 'blue', opacity: 5 }, fill: { color: 'pink' }, stroke: { color: 'green' } });
+  }
+  updateStyle(subProcessChildrenIds, { fill: { color: 'swimlane' }, stroke: { color: 'swimlane' }, font: { color: 'swimlane' } });
 
-  // Update text area
-  const textHeader = `Found ${elementsByKinds.length} ${bpmnKind}(s)`;
-  log(textHeader);
-  const lines = elementsByKinds.map(elt => `  - ${elt.bpmnSemantic.id}: '${elt.bpmnSemantic.name}'`).join('\n');
+  if (ShapeUtil.isBoundaryEvent(bpmnKind)) {
+    lastIdentifiedParentBpmnIds = getParentElementIds(otherIds);
+    updateStyle(lastIdentifiedParentBpmnIds, { opacity: 5, font: { color: 'green', opacity: 5 }, fill: { color: 'gold' }, stroke: { color: 'red' } });
+  }
 
-  textArea.value += [textHeader, lines].join('\n') + '\n';
-  textArea.scrollTop = textArea.scrollHeight;
+  const style = computeDefaultStyleByKind(bpmnKind);
+  updateStyle(otherIds, style);
```

Comment:
**suggestion:** inline the style constant

Reply:
I find less readable 2 function callings inlined.